### ISSUE TITLE
Error 'No node with name ... available' 

### ISSUE THREAD

**Issue by michielbicker** [RubendeBruin/DAVE#62]
Ha Ruben,

I get the following error when I try to open a model in DAVE:

![image](https://user-images.githubusercontent.com/89965911/133059413-93999f1e-d8a3-49a7-96e0-ddaa243c4eab.png)

I think it is because I duplicated shackles. In that process the names of the child objects are changed in a way that is different than you would expect if you insert a new shackle. Can I easily solve this? I can email the complete code if required.

Thanks!

**Comment by RubendeBruin** [comment]
Hi @michielbicker , if you can mail me the model-file or attach it to this post then I can have a look at it.

**Comment by michielbicker** [comment]
Ha @RubendeBruin,

I solved it by manually replacing the 'missing' nodes by the names that would be generated by the _s.new_shackle()_ command. Now the file is working again, but I think duplicating the shackle resulted in the error.

**Comment by RubendeBruin** [comment]
Good to hear. 
If you find out how I can reproduce it then I can have a look at how solve it.

**Comment by michielbicker** [comment]
The situation occurs if you generate a shackle with a certain name, let's say _shackle_. The pin, bow and inside then get the names _shackle_pin_, _shackle_bow_ and _shackle_inside_. Then I duplicate the shackle, which then becomes _shackle_1_. The sub items become _shackle_1_pin_, etc. But if I then rename _shackle_1_ into _something_else_ the sub items are still called _shackle_1_pin_, etc. (While I am typing this I realize that the situation is caused by renaming a shackle, it has nothing to do with the duplication...) The error occurs when something is connected to _shackle_1_pin_, which doesn't exist anymore when the model is reloaded (saved and opened), because then _shackle_1_pin_ is automatically renamed to _something_else_pin_. Is this a bit clear?

**Comment by RubendeBruin** [comment]
@michielbicker , yes clear. Confirmed and can reproduce:

```
s = Scene()


s.new_shackle('Shackle')
s.new_shackle(name="GP250", kind = "GP250")
s.new_geometriccontact("GP250" + "_connection", "GP250_pin", "Shackle_inside", inside=True)

s['Shackle'].name = 'something_else'  # <-- rename

code = s.give_python_code() # <-- this produces invalid python code in version 1.7.0

s2 = Scene()
s2.run_code(code)  # this errors
```

thanks for digging this up.

**Comment by RubendeBruin** [comment]
Fix implemented in 1.7.1
